### PR TITLE
configure jest with enzyme 

### PR DESCRIPTION
**Pull request by rady-ben**
* add new jest config
* add enzyme config
* test the render of the FileForm component

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "babel-jest": "24.9.0",
     "babel-loader": "^8.0.6",
     "babel-plugin-istanbul": "6.0.0",
+    "check-prop-types": "^1.1.2",
```
Comment (ancientTexts-net) [inline package.json:70]: TODO: discuss with the team: 
Seems like a helpful idea to verify prop-types on PUBLIC libraries.

File: src/components/file/FileForm.js [hunk 2]
```diff
@@ -59,6 +59,9 @@ FileForm.propTypes = {
   submitText: PropTypes.string,
   /** Function run when submit button is clicked */
   onSubmit: PropTypes.func.isRequired,
+  branch: PropTypes.string,
+  filepath: PropTypes.string,
+  defaultContent: PropTypes.string,
```
Comment (ancientTexts-net) [inline src/components/file/FileForm.js:64]: NOTE: there were a few missing proptypes

**Comment by ancientTexts-net** [comment]
___
*[package.json, line 32 at r1](https://reviewable.io/reviews/unfoldingword/gitea-react-toolkit/105#-Mj2CGMh7tPPOXvAf55Q:-Mj2CGMh7tPPOXvAf55R:b-l9edbu) ([raw file](https://github.com/unfoldingword/gitea-react-toolkit/blob/24846a93830d8140c2b336e7e1a20e50e4ceb02f/package.json#L32)):*
> ```JSON
>     "test:unit": "NODE_ENV=test jest ./src/core --coverage && cat ./coverage/lcov.info | coveralls",
>     "test": "nyc --exclude-after-remap=false npm run test:e2e",
>     "testJestEnzyme": "jest --watch -c ./jest.enzyme.config.js",
> ```

For now, there are two ways to invoke jest.

But if this PR is going to be approved, maybe we can refactor into a single (combined) script

**Comment by ancientTexts-net** [comment]
___
*[__mocks__/fileMock.js, line 1 at r1](https://reviewable.io/reviews/unfoldingword/gitea-react-toolkit/105#-Mj2CguE2kJC31SkAd80:-Mj2CguE2kJC31SkAd81:bdlptsx) ([raw file](https://github.com/unfoldingword/gitea-react-toolkit/blob/24846a93830d8140c2b336e7e1a20e50e4ceb02f/__mocks__/fileMock.js#L1)):*
> ```JavaScript
> module.exports = 'test-file-stub';
> ```

see `identity-obj-proxy`

**Comment by ancientTexts-net** [comment]
___
*[src/components/file/FileForm.js, line 22 at r1](https://reviewable.io/reviews/unfoldingword/gitea-react-toolkit/105#-Mj2D1dt31Xi6sISzpUM:-Mj2D1dt31Xi6sISzpUN:b-r7hrgp) ([raw file](https://github.com/unfoldingword/gitea-react-toolkit/blob/24846a93830d8140c2b336e7e1a20e50e4ceb02f/src/components/file/FileForm.js#L22)):*
> ```JavaScript
>     <Paper 
>       style={{ marginBottom: '1em', padding: '1.3em' }}
>       data-test="component-fileForm"
> ```

Several `data-test` elements added. 👍

**Comment by ancientTexts-net** [comment]
___
*[__mocks__/FileForm.test.js, line 27 at r1](https://reviewable.io/reviews/unfoldingword/gitea-react-toolkit/105#-Mj2DCpC0bJ81GMiRDhx:-Mj2DCpC0bJ81GMiRDhy:b-6hqune) ([raw file](https://github.com/unfoldingword/gitea-react-toolkit/blob/24846a93830d8140c2b336e7e1a20e50e4ceb02f/__mocks__/FileForm.test.js#L27)):*
> ```JavaScript
> const findByAttribute = (wrapper, attribute) => wrapper.find(`[data-test='${attribute}']`)
> 
> // render tests
> ```

TODO: can we look at these tests?

They look helpful to me.
- is this a good pattern for tests in general?
- do they offer meaningful coverage of statements?
- are they specific enough?
- do they cover possible edge cases?

**Review by ancientTexts-net — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword/gitea-react-toolkit/105)* status: 0 of 6 files reviewed, 5 unresolved discussions (waiting on @rady-ben)

**Comment by rady-ben** [comment]
> combined



> _[package.json, line 32 at r1](https://reviewable.io/reviews/unfoldingword/gitea-react-toolkit/105#-Mj2CGMh7tPPOXvAf55Q:-Mj2CGMh7tPPOXvAf55R:b-l9edbu) ([raw file](https://github.com/unfoldingword/gitea-react-toolkit/blob/24846a93830d8140c2b336e7e1a20e50e4ceb02f/package.json#L32)):_
> 
> > ```json
> >     "test:unit": "NODE_ENV=test jest ./src/core --coverage && cat ./coverage/lcov.info | coveralls",
> >     "test": "nyc --exclude-after-remap=false npm run test:e2e",
> >     "testJestEnzyme": "jest --watch -c ./jest.enzyme.config.js",
> > ```
> 
> For now, there are two ways to invoke jest.
> 
> But if this PR is going to be approved, maybe we can refactor into a single (combined) script

I added the new script because I saw this error message when I run the existing script error Command failed with exit code 1. and the script stops and exits. and there in no watch, jest doesn't refreshes when I updated the tests that’s why I had the idea to add new command with the new tests.
an other reason for separating is that the existing test concern the differents APIs, however the news ones concern the react components

**Review by jincypjose — APPROVED** [review]

File: __mocks__/FileCard.test.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+});
+
+test('switch the preview state', () => {
+    const wrapper = setupWrapper();
+    const blockEditable = findByAttribute(wrapper, 'blockEditable');
+    expect(blockEditable.props().preview).toBeTruthy();
+    const previewButton = findByAttribute(wrapper, 'previewButton');
+    previewButton.simulate('click');
+    expect(mockSetPreview).toBeCalledWith(false);
+    // NEED HELP
```
Comment (rady-ben) [inline __mocks__/FileCard.test.js:134]: @ancientTexts-net 
in this function I am testing the initial value to be `true` and the and when the user click on the button I test that that the `setPreview` is call with the `false` as param. which describe the switch from true to false. 
but I don't know how to set realy the state on the component to  `false` so I will be able to test the switch from false to true.
I found that in a class component we use something like this 
  `wrapper.instance().setPreview(false)`
I am looking for an equivalent for the hooks

File: src/components/file/FileCard.js [hunk 4]
```diff
@@ -40,25 +40,35 @@ function FileCard({
...
         <Paper>
           <BlockEditable
             preview={preview}
             markdown={markdown}
+            // NEED HELP
             onEdit={setMarkdown}
-            editable={!!authentication}
+            editable={isAuthenticated}
+            data-test="blockEditable"
           />
```
Comment (rady-ben) [inline src/components/file/FileCard.js:61]: @ancientTexts-net 
how can I simulate the onEdit event with just `BlockEditable.simulate('onEdit', mock Event);` doesn't work

File: src/components/file/FileCard.js [hunk 5]
```diff
@@ -40,25 +40,35 @@ function FileCard({
...
       <CardContent>
         <Paper>
           <BlockEditable
             preview={preview}
             markdown={markdown}
+            // NEED HELP
             onEdit={setMarkdown}
-            editable={!!authentication}
+            editable={isAuthenticated}
+            data-test="blockEditable"
```
Comment (rady-ben) [inline src/components/file/FileCard.js:60]: @ancientTexts-net
how can I simulate the onEdit event with just BlockEditable.simulate('onEdit', mock Event); doesn't work
